api/src/opentrons/protocol_engine/state/labware.py

```diff
@@ -190,9 +199,18 @@ def get(self, labware_id: str) -> LoadedLabware:
                 f"Labware {labware_id} not found."
             ) from e
 
-    def get_id_by_module(self, module_id: str) -> Optional[str]:
+    def get_id_by_module(self, module_id: str) -> str:
         """Return the ID of the labware loaded on the given module."""
-        raise NotImplementedError()
+        for (key, value) in self.state.labware_by_id.items():
```

Comment:
Parentheses unnecessary here. Also, more explicit variable names would read better here
```suggestion
        for labware_id, labware in self.state.labware_by_id.items():
```